File: src/main/scala/org/nygenome/als/graphdb/value/AbstractUniProtDrug.scala
package org.nygenome.als.graphdb.value

import java.util
import java.util.List

abstract class AbstractUniProtDrug extends ValueTrait{
  def drugModelType: String 
  def id: String 
  def name: String 
  def geneName: String 
  def genbankProteinId: String 
  def genbankGeneId: String 
  def uniprotId: String 
  def uniprotTitle: String 
  def pdbId: String 
  def geneCardId: String 
  def geneAtlasId: String 
  def hgncId: String 
  def species: String 
  def drugIdList: util.List[String] 

}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/AbstractUniProtDrug.kt
package org.biodatagraphdb.alsdb.model

/**
 * Created by fcriscuo on 2/3/20.
 */
 data class AbstractUniProtDrug(
    val drugModelType: String,
    val id: String,
    val name: String,
    val geneName: String,
    val genbankProteinId: String,
    val genbankGeneId: String,
    val uniprotId: String,
    val uniprotTitle: String,
    val pdbId: String,
    val geneCardId: String,
    val geneAtlasId: String,
    val hgncId: String,
    val species: String,
    val drugIdList: List<String>

) {
}